qubesmanager/global_settings.py

```diff
@@ -169,12 +169,20 @@ def __apply_system_defaults__(self):
 
         # default netvm
         if utils.did_widget_selection_change(self.default_netvm_combo):
-            try:
-                self.qubes_app.default_netvm = \
-                    self.default_netvm_combo.currentData()
-            except exc.QubesException as ex:
+            new_default_netvm = self.default_netvm_combo.currentData()
+            if new_default_netvm.property_is_default('netvm'):
                 self.errors.append(
-                    "Failed to set Default NetVM due to {}".format(str(ex)))
+                    'Failed to set Default NetVM: qube {} has its own NetVM '
+                    'set to "default". To avoid cycles, please change NetVM'
+                    'of qube {} to a non-default qube.'.format(
```

Comment:
How about:

> Cannot set {} as the default net qube. Reason: {}'s net qube is already set to "default," and a qube cannot be its own net qube. Please change {}'s net qube and try again.

(Let's please stop using "VM," and especially "NetVM," in user-facing text.)